Ports Nooses to bee so you can hang people

ro5490 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Ported from Hippie - https://github.com/HippieStation/HippieStation/pulls?q=is%3Apr+noose

Adds nooses, not much else to say

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Hang people for their crimes!

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: Added Nooses
soundadd: added sound files for the noose
imageadd: added icons for the noose
fix: Take a cable coil, stand on a chair, build a noose, then drag whoever onto it to hang them
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

ro5490 committed:
Merge pull request #1 from BeeStation/master

Updates to 513.1517 (#1400)

ro5490 committed:
Merge pull request #4 from BeeStation/master

Update to Master

ro5490 committed:
Adds Nooses to Bee
adds noose code to head.dm
adds noose sound files
adds noose.dm
adds noose crafting to cable coils

mehmetipekpatates commented:
epic

ike709 commented on code/game/objects/noose.dm:
```diff
@@ -0,0 +1,153 @@
+/obj/item/stack/cable_coil/building_checks(datum/stack_recipe/R, multiplier)
+	if(R.title == "noose")
```
Bad.

ro5490 replied:
Not sure what exactly about this section of code is bad
`/obj/item/stack/proc/building_checks(datum/stack_recipe/R, multiplier)` is used in stack.dm already and this just specifies the cable_coil checks
and `if(R.title =="noose")` seems alright to me for defining the noose recipe

ike709 replied:
String comparisons are slow. Check if `result_type` is the noose typepath instead.

ro5490 replied:
`if(R.result_type(noose)) ?`
cant find many references to `result_type` in the code, only in stacks.dm

ike709 replied:
`if(R.result_type == /whatever/noose/path/is)`

ro5490 replied:
ahh cool, Will fo that now

ike709 commented on code/game/objects/noose.dm:
```diff
@@ -0,0 +1,153 @@
+/obj/item/stack/cable_coil/building_checks(datum/stack_recipe/R, multiplier)
+	if(R.title == "noose")
+		if(!(locate(/obj/structure/chair) in usr.loc) && !(locate(/obj/structure/bed) in usr.loc) && !(locate(/obj/structure/table) in usr.loc) && !(locate(/obj/structure/toilet) in usr.loc))
```
Bad.

ro5490 replied:
```
/obj/item/stack/cable_coil/building_checks(datum/stack_recipe/R, multiplier)
	if(R.title == "noose")
		if(!(locate(/obj/structure/chair) in usr.loc))
			to_chat(usr, "<span class='warning'>You have to be standing on top of a chair to make a noose!</span>")
			return FALSE
	return ..()

```
bit better?

hotimep replied:
toliet noose

ike709 commented:
Didn't have time to fully review.

I'm *sure* nothing bad will come of this.

ro5490 commented:
> Didn't have time to fully review.
> 
> I'm _sure_ nothing bad will come of this.

Not exactly sure whats bad about these, so if someone could explain I'll happily try and optimise/fix it

ro5490 committed:
adjusts chair check code

ike709 commented on code/game/objects/noose.dm:
```diff
@@ -0,0 +1,153 @@
+/obj/item/stack/cable_coil/building_checks(datum/stack_recipe/R, multiplier)
+	if(R.title == "noose")
+		if(!(locate(/obj/structure/chair) in usr.loc))
```
```suggestion
		if(!(locate(/obj/structure/chair) in get_turf(usr)))
```

ike709 commented on code/game/objects/noose.dm:
```diff
@@ -0,0 +1,153 @@
+	icon = 'icons/obj/objects.dmi'
+	layer = FLY_LAYER
+	flags_1 = NODECONSTRUCT_1
+	var/image/over
```
Why is this not a `mutable_appearance`?

ro5490 replied:
Which line? I'm still not too familiar with byondcode tbh

ike709 replied:
`var/image/over`

ro5490 replied:
not sure, was direct ported from hippie, ill fix that though

ro5490 committed:
Update code/game/objects/noose.dm

Co-Authored-By: ike709 <ike709@users.noreply.github.com>

ike709 commented on code/game/objects/noose.dm:
```diff
@@ -0,0 +1,153 @@
+				playsound(buckled_mob.loc, 'sound/effects/noose_idle.ogg', 30, 1, -3)
+
+/mob/living/carbon/human
+	var/noosed = FALSE
```
No. We don't need a `human` var to see if someone is being hanged. Just check if they're in the structure's `buckled_mobs` or not to keep track of this.

ro5490 replied:
so, change var/noosed to a buckled mobs check instead, got it

ro5490 replied:
>removed this and the `checknoosedrop()`proc itself as its not needed it seems

ro5490 replied:
also updated code to have buckled mob check instead of ifnoosed

ike709 commented on code/modules/surgery/bodyparts/head.dm:
```diff
@@ -141,6 +141,7 @@
 	else if(!animal_origin)
 		var/mob/living/carbon/human/H = C
 		var/datum/species/S = H.dna.species
+		H.checknoosedrop()
```
Why are you calling this in update_limb()?

ro5490 replied:
not exactly sure, was direct ported from hippiecode, ill look into it, any suggestions here?

ro5490 replied:
>removed this and the proc itself as its not needed it seems

ike709 commented on code/modules/power/cable.dm:
```diff
@@ -9,6 +9,11 @@ GLOBAL_LIST_INIT(cable_colors, list(
 	))
 
+
+/obj/item/stack/cable_coil/Initialize(mapload, new_amount = null, param_color = null)
```
This incorrect usage of Initialize() is unnecessary.

ro5490 replied:
How do I go about fixing this?

ike709 replied:
Remove it. It's unneeded.

ro5490 replied:
so 
```
/obj/item/stack/cable_coil/Initialize(mapload, new_amount = null, param_color = null)
	.=..()
	recipes = list(new/datum/stack_recipe("cable restraints", /obj/item/restraints/handcuffs/cable, 15), new/datum/stack_recipe("noose", /obj/structure/chair/noose, 30, time = 80, one_per_turf = 1, on_floor = 1))

```
to 
```
/obj/item/stack/cable_coil/Initialize()
	.=..()
	recipes = list(new/datum/stack_recipe("cable restraints", /obj/item/restraints/handcuffs/cable, 15), new/datum/stack_recipe("noose", /obj/structure/chair/noose, 30, time = 80, one_per_turf = 1, on_floor = 1))
```
Or just delete the whole block?

ike709 replied:
The whole block

ike709 commented on code/game/objects/noose.dm:
```diff
@@ -0,0 +1,153 @@
+
+/mob/living/carbon/human/proc/checknoosedrop()
+	if(noosed)
+		for(var/obj/structure/chair/noose/noose in loc)
```
```suggestion
		for(var/obj/structure/chair/noose/noose in get_turf(src))
```

ro5490 committed:
Update code/game/objects/noose.dm

Co-Authored-By: ike709 <ike709@users.noreply.github.com>

ro5490 committed:
noose.dm gets mutable_appearance and Result_type
Also removes all references of "Noose ="
And removed checknoosedrop() Proc
Checks if mob has NOBREATH trait to call for oxyloss
fixes untying someone not dropping them to the floor

cable.dm adjusted init

Head.DM H.checknoosedrop() call has been removed

ro5490 committed:
removes unneeded recipie init from cable.dm

ro5490 commented:
Ik the whole visible message section in noose.dm is a mess, tried to fix it and got nowhere cause i got confused for a hot sec, Will continue to fiddle with it but if anyone can fix it up for me or give me pointers I'd appreciate it

Also added a trait check so it doesnt try and oxyloss mobs with no_breath trait

ro5490 committed:
Fixes text-chat shit in noose.dm

ro5490 committed:
Force checks

qwertyquerty commented on code/game/objects/noose.dm:
```diff
@@ -0,0 +1,142 @@
+				var/flavor_text = list("<span class='suicide'>[buckled_mob]'s legs flail for anything to stand on.</span>",\
+										"<span class='suicide'>[buckled_mob]'s hands are desperately clutching the noose.</span>",\
+										"<span class='suicide'>[buckled_mob]'s limbs sway back and forth with diminishing strength.</span>")
+				if(buckled_mob.stat == DEAD)
```
Take out the messages for if they are dead. It will just be a spam. Only send a message if they aren't dead.

ro5490 committed:
Noose flavortext only when buckled is alive
